src/config.rs

```diff
@@ -490,11 +485,14 @@ impl Cfg {
                 }
 
                 self.settings_file.with_mut(|s| {
-                    DEFAULT_METADATA_VERSION.clone_into(&mut s.version);
+                    s.version = MetadataVersion::default();
                     Ok(())
                 })
             }
-            _ => Err(RustupError::UnknownMetadataVersion(current_version).into()),
+            MetadataVersion::V12 => {
```

Comment:
To me it's clear that the in the original code the happy path is for `MetadataVersion::default()`, but now it's for `MetadataVersion::V12`. I know at the moment being it's all the same, but the meaning being conveyed seems to be a bit different 🤔

Reply:
One of the advantages of using an enum for things like this is exhaustive matching. If I add rewrite this arm to match like `version if version == MetadataVersion::default()`, the compiler is not smart enough to figure out this is still an exhaustive match. However if we ever add another variant to `MetadataVersion` this `match` will fail to compile anyway, and it should be pretty straightforward for someone to figure out the required change. To make it a little more obvious, I'll just add a comment explaining the context a little more.